web_client/src/store/index.ts

```diff
@@ -227,9 +229,11 @@ function progressHandler(completed, total) {
 }
 
 /** Creates array of tasks to run then runs tasks in parallel. */
-function startReaderWorkerPool() {
+function startReaderWorkerPool(onDownloadProgress) {
   // Get the current array of tasks in readDataQueue
-  const taskArgsArray = readDataQueue.map((taskInfo) => [taskInfo]);
+  const taskArgsArray = readDataQueue.map((taskInfo) => [
+    Object.assign(taskInfo, { onDownloadProgress }),
```

Comment:
Should we do the spread operator here, or do we have to do `Object.assign` to modify in-place?